kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts

```diff
@@ -92,52 +92,117 @@ export class OnEnter extends OnUUIDEvent<'onEnter'> {
 /**
  * @public
  */
-@Component('engine.onChange', CLASS_ID.UUID_CALLBACK)
-export class OnChanged extends OnUUIDEvent<'onChange'> {
+export type OnPointerUUIDEventOptions = {
+  button?: ActionButton
+  hoverText?: string
+  showFeeback?: boolean
+  distance?: number
+}
+
+/**
+ * @public
+ */
+@Component('engine.onClick', CLASS_ID.UUID_CALLBACK)
+export class OnClick extends OnPointerUUIDEvent<'onClick'> {
   @ObservableComponent.readonly
-  readonly type: string = 'onChange'
-  constructor(callback: (event: IEvents['onChange']) => void) {
+  readonly type: string = 'onClick'
+
+  constructor(callback: (event: IEvents['onClick']) => void)
+  constructor(callback: (event: IEvents['onClick']) => void, options: OnPointerUUIDEventOptions)
+  constructor(callback: (event: IEvents['onClick']) => void, options?: any) {
```

Comment:
To be consistent with the implementation of `OnPointerDown`.
```suggestion
  constructor(callback: (event: IEvents['onClick']) => void, options?: OnPointerUUIDEventOptions) {
```